Modules/Core/SpatialObjects/test/itkPolygonSpatialObjectIsInsideInObjectSpaceTest.cxx

```diff
@@ -0,0 +1,150 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkPolygonSpatialObject.h"
+#include <itkGroupSpatialObject.h>
+#include <itkSpatialObjectToImageFilter.h>
+#include <itkImageFileReader.h>
+#include <itkImageFileWriter.h>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+// -------------------------------------------------------------------------------------
+// The purpose of this test is to make sure that the function
+// ObjectIsInsideInObjectSpace is working as expected.
+// This is related to a regression that was fixed under case 1082
+// which showed that certain contours that were created from the polygonSpatialObject
+// had trailing tails that were not supposed to be there.
+// In order to test this we take a known regression list of contour points
+// and create a contour image from these points.
+// We compare the resulting image with the known baseline contour image.
+// ---------------------------------------------------------------------------------------
+
+using PointType = itk::ImageBase<2>::PointType;
+using PolygonPointType = itk::PolygonSpatialObject<2>::PolygonPointType;
+using PolygonPointListType = itk::PolygonSpatialObject<2>::PolygonPointListType;
+using PixelType = uint16_t;
+using Group = itk::GroupSpatialObject<2>;
+using Image2D = itk::Image<PixelType, 2>;
+using SpatialObjectToImageFilterType = itk::SpatialObjectToImageFilter<Group, Image2D>;
+
+PolygonPointListType
+getPolygonPointsFromCSVFile(const std::string & iCSVFile)
+{
+  PolygonPointListType returnList;
+  std::ifstream        fs;
+  char                 data[100];
+
+  fs.open(iCSVFile);
+  if (fs.good())
+  {
+    // read over first line
+    fs.getline(data, sizeof(data), '\n');
+    char commaDelim;
+    while (fs.getline(data, sizeof(data), '\n'))
+    {
+      std::stringstream ss(data);
+
+      PolygonPointType               polygonPt;
+      itk::SpatialObject<2>::Pointer so = itk::SpatialObject<2>::New();
+      polygonPt.SetSpatialObject(so);
+      PointType pt;
+
+      ss >> pt[0];
+      ss >> commaDelim;
+      ss >> pt[1];
+
+      polygonPt.SetPositionInWorldSpace(pt);
+      returnList.push_back(polygonPt);
+    }
+  }
+  else
+  {
+    std::cout << "error reading csv file : " << iCSVFile << std::endl;
+  }
+  fs.close();
+  return returnList;
+}
+
+Group::Pointer
+createSpatialGroupFromPolygonPoints(const PolygonPointListType & i_polygonPointList)
+{
+
+  auto groupPtr = Group::New();
+  auto polygonPtr = itk::PolygonSpatialObject<2>::New();
+  groupPtr->AddChild(polygonPtr);
+  polygonPtr->SetPoints(i_polygonPointList);
+  polygonPtr->SetDefaultInsideValue(1.0);
+  polygonPtr->SetDefaultOutsideValue(0.0);
+  polygonPtr->Update();
+  return groupPtr;
+}
+
+Image2D::Pointer
+readImageFile(const std::string & iFileName)
```

Comment:
We usually do not define a helper method for reading/writing images. We also avoid tray/catch block boilerplate code using the `ITK_TRY_EXPECT_NO_EXCEPTION()` macro when updating the reader (and in general any filter) (there are numerous examples throughout the existing tests).

We would also use it when updating the writer.

Also, taking into account that the test's main body is short, I would also avoid using the helper methods above. I think the body of the test would still be relatively light, would read well and would also free us from adding template arguments to them.

Reply:
Addressed.